src/Pages/Admin/Configuration.tsx

```diff
@@ -35,66 +35,54 @@ const getWindowCenterPosition = (windowWidth, windowHeight) => {
   return { left, top };
 };
 
-class Configuration extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      currentProjectInputValue: this.props.cookies.get(
-        selectedProjectCookieName
-      ),
-    };
-
-    this.handleProjectInputChange = this.handleProjectInputChange.bind(this);
-    this.handleSetProjectSubmit = this.handleSetProjectSubmit.bind(this);
-    this.setNewProjectId = this.setNewProjectId.bind(this);
-    this.receiveMessage = this.receiveMessage.bind(this);
-    this.waitUntilProjectAccessible = this.waitUntilProjectAccessible.bind(
-      this
-    );
-  }
+const Configuration: React.FC = () => {
+  const cookies = new Cookies(document.cookie);
+  const cookie = cookies.get(selectedProjectCookieName);
+  const navigate = useNavigate();
+  const [currentProjectInputValue, setCurrentProjectInputValue] = useState(cookie);
 
-  componentDidMount() {
-    window.addEventListener('message', this.receiveMessage, false);
-  }
 
-  componentWillUnmount() {
-    window.removeEventListener('message', this.receiveMessage);
-  }
+  useEffect(() => {
+    window.addEventListener('message', receiveMessage, false)
+
+    // clean up the event every time the component is re-rendered
+    return function cleanup() {
+      window.removeEventListener('message', receiveMessage)
+    }
+  })
 
-  handleProjectInputChange(event) {
-    this.setState({ currentProjectInputValue: event.target.value });
+  const handleProjectInputChange = (event: ChangeEvent<HTMLInputElement>): void =>  {
+    setCurrentProjectInputValue(event.target.value);
   }
 
-  handleSetProjectSubmit(event) {
+  const handleSetProjectSubmit = (event: any): void => {
```

Comment:
event does not have a type?

Reply:
[9ede5d4](https://github.com/Kentico/kontent-sample-app-react/pull/206/commits/9ede5d4e8127366824a28d497840cf0cf5e1cc2e)